# ドキュメントの配置構造

**更新日**: 2025/3/24
**確認日**: 2025/3/24

## 目的

- ドキュメントの配置構造を定義し、人とLLMがドキュメントを発見しやすくする
- 対象読者別のアクセスパスを最適化し、必要な情報への効率的なアクセスを実現する
- 時間軸（現在の機能と未来の機能）による分類を明確にする
- ルールとコンテンツを分離し、管理の一貫性を確保する

## 配置構造の基本原則

1. **対象者中心**: 主要な情報を開発者・運用者・エンドユーザーなど対象読者別に整理
2. **明確な責任分離**: 開発、運用、利用の各領域を明確に区分
3. **ルールとコンテンツの分離**: 規約と実際のコンテンツを分けて管理
4. **時間軸による分類**: 現在の機能と未来の機能を区別
5. **適切な粒度**: 情報を適切な粒度に分割し、単一の責任原則を適用
6. **アクセシビリティ**: 重要な情報へのアクセスパスを短くし、発見可能性を高める

## git-resumeプロジェクトのドキュメント構造

git-resumeプロジェクトでは、以下のディレクトリ構造を採用し、ドキュメントを組織化します。

### ルートディレクトリ

- `README.md`: プロジェクトの概要説明、主要機能、インストール方法
- `.env.sample`: 環境変数の例と説明
- `.clinerules/`: Cline拡張機能のルールとガイドライン
- `docs/`: すべてのドキュメント
- `apps/`: すべてのアプリケーション
  - `api/`: APIサーバーアプリケーション
    - `README.md`: API実装の詳細と使用方法
  - `cli/`: コマンドラインインターフェースアプリケーション
    - `README.md`: CLIツールの詳細と使用方法
  - `web/`: Webアプリケーション
    - `README.md`: Webアプリケーションの詳細と使用方法
- `packages/`: 共有パッケージ（モノレポ構造）
  - `models/`: データモデル定義
    - `README.md`: データモデルの説明と使用方法
  - `services/`: 共通サービス
    - `README.md`: 共通サービスの説明と使用方法
  - `utils/`: ユーティリティ関数
    - `README.md`: ユーティリティ関数の説明と使用方法
- `infra/`: インフラストラクチャ関連ファイル
  - `README.md`: インフラストラクチャの説明と使用方法
  - `environments/`: 環境別設定

### docs/ ディレクトリ構造

```
docs/
├── README.md                         # ドキュメントガイドライン
├── PLAN_AND_PROGRESS.md              # ドキュメントルールの適用計画と進捗状況
├── guide/                            # すべてのガイドのIndex
│   ├── README.md                     # ガイド全体の目次と導入
│   ├── developer/                    # 開発者向けガイド
│   │   ├── README.md                 # 開発者向けガイド目次
│   │   ├── getting-started/          # 開発環境構築等
│   │   │   └── README.md
│   │   ├── development-flow/         # 開発フロー関連
│   │   │   └── README.md
│   │   ├── packages/                 # モノレポ内のパッケージ概要
│   │   │   └── README.md
│   │   ├── architecture/             # アーキテクチャ関連
│   │   │   ├── adr/                  # アーキテクチャ決定記録
│   │   │   └── README.md
│   │   │   └── ...
│   │   └── ...
│   ├── operator/                     # 運用者向けガイド
│   │   ├── README.md                 # 運用者向けガイド目次
│   │   ├── deployment/               # デプロイメント関連
│   │   │   └── README.md
│   │   ├── monitoring/               # 監視関連
│   │   │   └── README.md
│   │   ├── maintenance/              # メンテナンス関連
│   │   │   └── README.md
│   │   └── ...
│   └── users/                        # ユーザー向けガイド
│       ├── README.md                 # ユーザー向けガイド目次
│       ├── basics/                   # 基本操作ガイド
│       │   └── README.md
│       ├── integration/              # 連携関連ガイド
│       │   └── README.md
│       └── ...
├── templates/                        # すべてのテンプレートを集約
│   ├── README.md                     # テンプレート全体の説明と使い方
│   ├── documents/                    # ドキュメント作成用テンプレート
│   │   ├── prd-template.md           # 製品要求仕様書 (PRD) テンプレート
│   │   ├── release-note-template.md  # リリースノートテンプレート
│   │   ├── retrospective-template.md # 振り返りレポートテンプレート
│   │   ├── tdd-template.md           # 技術設計書 (TDD) テンプレート
│   │   ├── test-plan-template.md     # テスト計画テンプレート
│   │   └── meeting-minutes-template.md # 会議議事録テンプレート
│   ├── development/                  # 開発用テンプレート (今後作成予定)
│   └── operations/                   # 運用用テンプレート (今後作成予定)
├── rules/                            # ドキュメントルール
│   ├── README.md                     # ドキュメントルール一覧と概要
│   ├── ai/                           # AI関連ドキュメントルール (今後作成予定)
│   │   ├── README.md                 # AI関連ルールの概要
│   │   ├── config-files.md           # AIの設定ファイル確認
│   │   ├── commit.md                 # コミットメッセージルール
│   │   ├── collaboration.md          # AIとの協働ルール
│   │   ├── minimum-change.md         # 最小変更ルール (追加予定)
│   │   ├── ask.md                    # 質問ルール (追加予定)
│   │   ├── maintenance.md            # ドキュメント更新ルール (追加予定)
│   │   ├── (削除済み)                # (LLM向けIndex - 統合済み)
│   │   ├── tools/                    # ツール活用ルール
│   │   │   ├── README.md             # ツール活用ルール一覧
│   │   │   └── repomix.md            # Repomix活用ルール
│   ├── guide/
│   │   ├── developer/                # 開発者向けドキュメントルール
│   │   │   ├── code/                 # コードファイルルール
│   │   │   ├── package/              # パッケージREADMEルール
│   │   │   ├── architecture/         # アーキテクチャルール
│   │   │   └── development-flow/     # 開発フローのルール
│   │   ├── operator/                 # 運用者向けルール
│   │   └── users/                    # ユーザー向けルール
│   ├── documents/                    # ドキュメント全般ルール
│   │   ├── code-doc-sync.md          # コードとドキュメントの同期ルール
│   │   ├── metadata.md               # メタデータルール
│   │   ├── navigation.md             # ナビゲーションルール
│   │   ├── README.md                 # 概要
│   │   └── relations.md              # 関連ドキュメント
│   ├── links.md                      # リンク集
│   ├── self-check/                   # ドキュメント検証システムの自己テスト
│   │   ├── README.md                 # 自己テストの説明と使用方法
│   │   └── test/                     # テストケース用ディレクトリ
│   │       ├── feature-spec/         # 機能仕様書の検証用ファイル
│   │       │   ├── correct.md        # 正しい形式の機能仕様書
│   │       │   └── wrong.md          # 間違った形式の機能仕様書
│   │       └── api-doc/              # API仕様書の検証用ファイル
│   │           ├── correct.md        # 正しい形式のAPI仕様書
│   │           └── wrong.md          # 間違った形式のAPI仕様書
│   ├── structure.md                  # ドキュメント構造
│   ├── types.md                      # ドキュメント種別
│   └── validation.md                 # ドキュメント検証ルール
└── product/                          # プロダクト情報
    ├── README.md                     # プロダクト概要
    ├── vision.md                     # ビジョン・ミッション
    ├── product-goal.md               # 製品目標
    ├── value-proposition.md          # 価値提案
    ├── members.md                    # 開発メンバー
    ├── features/                     # リリース済み機能
    │   └── README.md                 # 機能一覧と概要
    ├── roadmaps/                     # 未来の機能計画 (今後作成予定)
        ├── README.md                 # ロードマップ概要 (今後作成予定)
        └── [予定機能名]/             # 予定機能の詳細計画 (今後作成予定)
            ├── README.md             # 概要とドキュメント構造 (今後作成予定)
            ├── requirements/         # 非エンジニア向け (今後作成予定)
            │   ├── README.md         # 要件ドキュメントガイド (今後作成予定)
            │   ├── PRD.md            # 製品要件定義書 (今後作成予定)
            │   └── images/           # 図表・モックアップ (今後作成予定)
            └── systems/              # エンジニア向け (今後作成予定)
                ├── README.md         # システムドキュメントガイド (今後作成予定)
                ├── overview-specs/   # システム全体設計 (今後作成予定)
                │   ├── README.md     # 全体設計概要 (今後作成予定)
                │   └── system.md     # システム全体設計書 (今後作成予定)
                ├── package-specs/    # パッケージ仕様書 (今後作成予定)
                │   ├── README.md     # パッケージ仕様概要 (今後作成予定)
                │   ├── [パッケージ名]/ # 各パッケージ仕様 (今後作成予定)
                │   └── ...
                └── test-specs/       # テスト仕様書 (今後作成予定)
                    ├── README.md     # テスト戦略概要 (今後作成予定)
                    ├── e2e-tests.md  # E2Eテスト仕様 (今後作成予定)
                    └── unit-tests.md # ユニットテスト仕様 (今後作成予定)
```

### アプリケーションとパッケージのREADME

各アプリケーションとパッケージのディレクトリには、それぞれ専用のREADMEファイルを配置します：

- `apps/api/README.md`: API実装の詳細と使用方法
- `apps/cli/README.md`: CLIツールの詳細と使用方法
- `apps/web/README.md`: Webアプリケーションの詳細と使用方法
- `packages/models/README.md`: データモデルの説明と使用方法
- `packages/services/README.md`: 共通サービスの説明と使用方法
- `packages/utils/README.md`: ユーティリティ関数の説明と使用方法

### .clinerules ディレクトリ構造

```
.clinerules/
├── README.md                         # .clinerules全体の目次
├── important-docs.md                 # 重要ドキュメント一覧
├── commit-rules.md                   # プロンプト履歴の記録ルール
├── repomix.md                        # リポジトリコード効率的理解の方法
└── prompts/                          # プロンプトテンプレート
    └── update-docs.md                # ドキュメント更新時のルールとテンプレート
```

## 対象読者別のナビゲーションパス

効率的な情報アクセスのために、主要な対象読者別のナビゲーションパスを以下のように定義します：

### 新規開発者向けパス

1. `README.md` → プロジェクト概要の把握
   - 主要機能、技術スタック、インストール方法の確認
2. `docs/guide/developer/README.md` → 開発者ガイドの目次
   - 開発者向けドキュメントの全体像を把握
3. `docs/guide/developer/getting-started/setup.md` → 開発環境のセットアップ
   - 必要なツール、依存関係、環境変数の設定
4. `docs/guide/developer/architecture/overview.md` → システムアーキテクチャの理解
   - コンポーネント構成、データフロー、技術選定の理由
5. `docs/guide/developer/development-flow.md` → 開発プロセスの理解
   - ブランチ戦略、コードレビュー、CI/CDパイプライン

### 運用者向けパス

1. `README.md` → プロジェクト概要の把握
   - 主要機能、技術スタックの確認
2. `docs/guide/operator/README.md` → 運用者ガイドの目次
   - 運用に関するドキュメントの全体像を把握
3. `docs/guide/operator/deployment/README.md` → デプロイメント手順
   - 環境別のデプロイ方法、設定パラメータ
4. `docs/guide/operator/monitoring/README.md` → 監視方法
   - メトリクス、アラート、ログ収集の設定
5. `docs/guide/operator/maintenance/README.md` → メンテナンス手順
   - バックアップ、復旧、アップデート手順

### エンドユーザー向けパス

1. `README.md` → 基本的な概要と機能の把握
   - 主要機能、利用開始方法の確認
2. `docs/guide/users/README.md` → ユーザーガイドの目次
   - ユーザー向けドキュメントの全体像を把握
3. `docs/guide/users/basics/README.md` → 基本操作ガイド
   - 初期設定、基本的な使い方
4. `docs/guide/users/features/README.md` → 特定機能の詳細な使用方法
   - 機能別の詳細な操作手順
5. `docs/guide/users/troubleshooting.md` → 問題が発生した場合の対処法
   - よくある問題と解決方法、サポート連絡先

### LLM向けアクセスパス

1. `.clinerules` → 重要ドキュメントの一覧
   - プロジェクト全体を理解するための主要ドキュメント
2. `docs/README.md` → ドキュメンテーションシステムのガイドライン
   - ドキュメント体系の理解
3. `docs/rules/README.md` → ドキュメントルールの概要
   - ドキュメント作成・管理のルール一覧
4. `docs/rules/structure.md` → プロジェクト全体の構造
   - 何かを探すときに利用する

## ファイルとディレクトリの命名規則

### ディレクトリ名

- 小文字のケバブケース（ハイフン区切り）を使用
- 短く、明確で、内容を表す名前を選択
- 複数形を使用（例: `features/`, `rules/`）

### ファイル名

- 小文字のケバブケース（ハイフン区切り）を使用
- 大文字の例外: `README.md`, `CHANGELOG.md`, `PROGRESS.md`
- 内容を明確に表す名前を選択
- 特定のコンポーネントに関するファイルの場合はコンポーネント名を接頭辞として使用

## ドキュメント間の相互参照

ドキュメント間の関連性を明確にし、ナビゲーションを容易にするために、以下の相互参照ルールを適用します：

### 基本ルール

1. **相対パスの使用**: 他のドキュメントへのリンクは相対パスを使用
   ```markdown
   [ドキュメントガイドライン](../../README.md)
   [ドキュメント種別](./types.md)
   ```

2. **リンクテキストの明確化**: リンクテキストはリンク先の内容を明確に示す
   ```markdown
   [ドキュメント配置構造](./structure.md) // 良い例
   [こちら](./structure.md) // 避けるべき例
   ```

3. **双方向リンク**: 緊密に関連するドキュメント間では双方向のリンクを維持
   - ドキュメントAがドキュメントBを参照する場合、ドキュメントBもドキュメントAを参照する

4. **「関連ドキュメント」セクション**: 各ドキュメントの末尾に関連ドキュメントへのリンク一覧を提供
   ```markdown
   ## 関連ドキュメント
   
   - [ドキュメントガイドライン](../../README.md) - ドキュメント管理の基本方針
   - [ドキュメント種別](./types.md) - ドキュメントの種類と目的
   ```

### パンくずリスト

各ドキュメントの冒頭に階層構造を示すパンくずリストを追加し、現在位置の把握と上位階層への移動を容易にします：

```markdown
**パス**: [ドキュメントガイド](../../README.md) > [ドキュメントルール](../README.md) > [配置構造](./structure.md)
```

### 関連ドキュメントセクションの例

```markdown
## 関連ドキュメント

### 主要関連ドキュメント
- [ドキュメントガイドライン](../../README.md) - ドキュメント管理の基本方針
- [ドキュメントルール](../README.md) - ドキュメントルールの一覧

### 関連するルール
- [ドキュメント種別](./types.md) - ドキュメントの種類と目的
- [ドキュメントリンク方法](./links.md) - リンク作成のルール

### 被参照ドキュメント
- [ドキュメントナビゲーション](./documents/navigation.md) - ナビゲーション構造の設計
```

## 新しいドキュメントの配置ガイドライン

新しいドキュメントを作成する際は、以下の手順に従って適切な配置場所を決定してください：

1. ドキュメントの主な目的と対象読者を特定する
2. [ドキュメント種別の定義](/docs/rules/types.md)を参照し、適切な種別を選択する
3. 選択した種別に対応する配置場所を本ドキュメントから特定する
4. 既存の階層構造と命名規則に従ってファイルを作成する
5. フォルダ内にREADME.mdしか配置しない場合は階層を浅くするために、上階層に[rule-name].mdとして作成する

## ドキュメント構造の拡張と変更

プロジェクトの進化に伴い、ドキュメント構造の拡張や変更が必要になる場合があります。その際は、以下のプロセスに従ってください：

1. 変更の目的と必要性を明確にする
2. 影響範囲を分析し、既存ドキュメントへの影響を評価する
3. チームメンバーとレビューし、フィードバックを収集する
4. 本ドキュメントを更新し、変更内容を反映する
5. 変更を全チームメンバーに周知する
6. 必要に応じて、既存ドキュメントの移行計画を策定し実行する

## AIインストラクション

このルールを読んだAIは以下の行動をとってください：

- 定義された構造に従ってドキュメントの配置を提案し、必要に応じて再構成を推奨する
- 対象読者別アクセスパス、相互参照、時間軸分類を考慮したナビゲーション設計を行う
- ルールとコンテンツの分離原則を維持し、適切な階層構造を確保する
- 新しいドキュメント作成時に、その種別と目的に応じた最適な配置場所を提案する
- ドキュメント間のリンクが切れている場合は修正を提案する

## メタデータ

**更新・確認情報**:
- 最終更新日: 2025/03/24 13:10
- 最終確認日: 2025/03/24

**文書情報**:
- ステータス: 承認済み
- バージョン: 1.1.0

## 関連ドキュメント

### 主要関連ドキュメント
- [ドキュメントガイドライン](../../README.md) - ドキュメント管理の基本方針
- [ドキュメントルール](../README.md) - ドキュメントルールの一覧
- [ドキュメントルールの適用計画と進捗状況](../../PLAN_AND_PROGRESS.md) - 実装計画と現在の進捗状況

### 関連するルール
- [ドキュメント種別](./types.md) - ドキュメントの種類と目的
- [ドキュメントリンク方法](./links.md) - リンク作成のルール
- [ドキュメントナビゲーション](./documents/navigation.md) - ナビゲーション構造の設計

### 被参照ドキュメント
- [ドキュメントガイドライン](../../README.md) - ドキュメント構造の概要説明で参照

## Changelog

- 2025/3/24: [更新] LLM向けIndexの削除を反映し、ディレクトリ構造を更新
- 2025/3/24: [強化] ドキュメント間の相互参照の例を追加し、関連ドキュメントセクションを強化
- 2025/3/24: [更新] PLAN.mdとPROGRESS.mdの参照をPLAN_AND_PROGRESS.mdに更新
- 2025/3/24: [強化] 対象読者別のナビゲーションパスの例をより具体的に
- 2025/3/23: [追加] ドキュメント検証システムの自己テスト（self-check）ディレクトリを追加
- 2025/3/23: [追加] メタデータと関連ドキュメントセクションを追加
- 2025/3/23: ドキュメント構造を最新化
- 2025/3/23: 初回作成